govc/library/deploy.go

```diff
@@ -168,35 +169,68 @@ func (cmd *deploy) Run(ctx context.Context, f *flag.FlagSet) error {
 			dsID = ds.Reference().Value
 		}
 
-		deploy := vcenter.Deploy{
-			DeploymentSpec: vcenter.DeploymentSpec{
-				Name:               name,
-				DefaultDatastoreID: dsID,
-				AcceptAllEULA:      true,
-				Annotation:         cmd.Options.Annotation,
-				AdditionalParams: []vcenter.AdditionalParams{
-					{
-						Class:       vcenter.ClassOvfParams,
-						Type:        vcenter.TypeDeploymentOptionParams,
-						SelectedKey: cmd.Options.Deployment,
+		cmd.FolderFlag.Log("Deploying library item...\n")
+
+		var ref *types.ManagedObjectReference
+
+		switch item.Type {
+		case "ovf":
+			deploy := vcenter.Deploy{
+				DeploymentSpec: vcenter.DeploymentSpec{
+					Name:               name,
+					DefaultDatastoreID: dsID,
+					AcceptAllEULA:      true,
+					Annotation:         cmd.Options.Annotation,
+					AdditionalParams: []vcenter.AdditionalParams{
+						{
+							Class:       vcenter.ClassOvfParams,
+							Type:        vcenter.TypeDeploymentOptionParams,
+							SelectedKey: cmd.Options.Deployment,
+						},
 					},
+					NetworkMappings:     networks,
+					StorageProvisioning: cmd.Options.DiskProvisioning,
+					StorageProfileID:    cmd.profile,
 				},
-				NetworkMappings:     networks,
-				StorageProvisioning: cmd.Options.DiskProvisioning,
-				StorageProfileID:    cmd.profile,
-			},
-			Target: vcenter.Target{
-				ResourcePoolID: rp.Reference().Value,
-				HostID:         hostID,
-				FolderID:       folder.Reference().Value,
-			},
-		}
-
-		cmd.FolderFlag.Log("Deploying library item...\n")
+				Target: vcenter.Target{
+					ResourcePoolID: rp.Reference().Value,
+					HostID:         hostID,
+					FolderID:       folder.Reference().Value,
+				},
+			}
+			ref, err = m.DeployLibraryItem(ctx, item.ID, deploy)
+			if err != nil {
+				return err
+			}
+		case "vm-template":
+			storage := &vcenter.DiskStorage{
+				Datastore: dsID,
+				StoragePolicy: &vcenter.StoragePolicy{
+					Policy: cmd.profile,
+					Type:   "USE_SOURCE_POLICY",
+				},
+			}
+			if cmd.profile != "" {
+				storage.StoragePolicy.Type = "USE_SPECIFIED_POLICY"
+			}
 
-		ref, err := m.DeployLibraryItem(ctx, item.ID, deploy)
-		if err != nil {
-			return err
+			deploy := vcenter.DeployTemplate{
+				Name:          name,
+				Description:   cmd.Options.Annotation,
+				DiskStorage:   storage,
+				VMHomeStorage: storage,
```

Comment:
Would it ever make sense for the DiskStorage and VMHomeStorage to be in separate locations?

Reply:
Maybe, but we don't support splitting them out in the soap based commands such as `vm.create` and `vm.clone`.  Never had a request to support that, so kept em the same here too.